src/Microsoft.Azure.SignalR.Management/ServiceManager.cs

```diff
@@ -158,13 +158,19 @@ public async Task<bool> IsServiceHealthy(CancellationToken cancellationToken)
         {
             var isHealthy = false;
             var api = _restApiProvider.GetServiceHealthEndpoint();
-            await _restClient.SendAsync(api, HttpMethod.Get, _productInfo, handleResponse: (request, response) =>
+            await _restClient.SendAsync(api, HttpMethod.Get, _productInfo, handleExpectedResponse: response =>
                 {
-                    if (response.StatusCode == HttpStatusCode.OK)
+                    switch (response.StatusCode)
                     {
-                        isHealthy = true;
+                        case HttpStatusCode.OK:
+                            isHealthy = true;
+                            return true;
+                        case HttpStatusCode.ServiceUnavailable:
```

Comment:
maybe 502

Reply:
I found 503 in [rest api doc](https://github.com/Azure/azure-signalr/blob/dev/docs/rest-api.md#Service-Health). Do you mean signalr service works as a proxy, so 502 is better?

Reply:
will update in another PR